src/enhance.ts

```diff
@@ -79,11 +87,14 @@ export const enhanceDoc = (chakraDoc: string = ''): Promise<string> => {
   const reactImports = new Set<string>();
   const spinnerImports = new Set<string>();
 
-  const enhanced = chakraDoc.replaceAll(codeRegex, (_, codeBlock) => {
+  const enhanced = chakraDoc.replace(
+    headerRegex, (headerBlock) => headerBlock && `# ${headerBlock.match(componentNameRegex)}`
+  ).replaceAll(new RegExp(ignoredComponentsRegex, 'gmi'), ''
+  ).replaceAll(codeRegex, (_, codeBlock) => {
     const components: string[] = uniq(
       [...codeBlock.matchAll(componentsRegex)].map(
         ([_, component]) => component
-      )
+      ).filter(component => !ignoredComponentList.includes(component))
```

Comment:
this should not be needed since you already removed the ignored components and this only looks in the code blocks